independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -536,17 +549,30 @@ public boolean test(MethodInfo method) {
             if (!parameters.isEmpty() && (beanArchiveIndex != null)) {
                 String originalClassPackage = DotNames.packageName(originalClazz.name());
                 for (Type type : parameters) {
-                    ClassInfo parameterClassInfo = beanArchiveIndex.getClassByName(type.name());
-                    if (parameterClassInfo == null) {
-                        continue; // hope for the best
+                    if (type.kind() == Kind.PRIMITIVE) {
+                        continue;
                     }
-                    if (Modifier.isPrivate(parameterClassInfo.flags())) {
-                        return true; // parameters whose class is private can not be loaded, as we would end up with IllegalAccessError when trying to access the use the load the class
+                    DotName typeName = type.name();
+                    if (type.kind() == Kind.ARRAY) {
+                        typeName = type.asArrayType().component().name();
+                    }
+                    ClassInfo param = beanArchiveIndex.getClassByName(typeName);
+                    if (param == null) {
+                        LOGGER.warn(String.format(
+                                "Parameter type info not available: %s - unable to validate the parameter type's visibility for method %s declared on %s",
+                                type.name(), method.name(), method.declaringClass().name()));
+                        continue;
                     }
-                    if (!Modifier.isPublic(parameterClassInfo.flags())) {
-                        // parameters whose class is package-private and the package is not the same as the package of the method for which we are checking can not be loaded,
-                        // as we would end up with IllegalAccessError when trying to access the use the load the class
-                        return !DotNames.packageName(parameterClassInfo.name()).equals(originalClassPackage);
+                    if (Modifier.isPublic(param.flags()) || Modifier.isProtected(param.flags())) {
+                        continue;
+                    }
+                    // e.g. parameters whose class is package-private and the package is not the same as the package of the method for which we are checking can not be loaded,
+                    // as we would end up with IllegalAccessError when trying to access the use the load the class
+                    if (!DotNames.packageName(param.name()).equals(originalClassPackage)) {
```

Comment:
IMHO this should be an error rather than a warning. We have seen cases where security interceptors are not applied even though the user was expecting them to be.

Reply:
I agree but the problem is that if you annotate a class with an interceptor binding then all methods (incl. inherited ones) should be intercepted and sometimes you need to extend a class from a different package which you don't control. Such an extended class may declare e.g. a package-private method that accepts a private param type. And since you're not able to modify the extended class you'd be screwed. An example from the [original issue](https://github.com/quarkusio/quarkus/issues/18477) is `java.util.concurrent.ThreadPoolExecutor` and `ThreadPoolExecutor.runWorker(Worker)`.

Reply:
I'm going to merge this PR. @stuartwdouglas if you really think that the current behavior is wrong (actually, before this PR the method is ignored and no warning is logged) then pls file a new issue.